x/stakeibc/keeper/redeem_stake_test.go

```diff
@@ -0,0 +1,308 @@
+package keeper_test
+
+import (
+	"fmt"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	epochtypes "github.com/Stride-Labs/stride/v22/x/epochs/types"
+	recordtypes "github.com/Stride-Labs/stride/v22/x/records/types"
+	stakeibctypes "github.com/Stride-Labs/stride/v22/x/stakeibc/types"
+)
+
+type RedeemStakeState struct {
+	epochNumber                        uint64
+	initialNativeEpochUnbondingAmount  sdkmath.Int
+	initialStTokenEpochUnbondingAmount sdkmath.Int
+}
+type RedeemStakeTestCase struct {
+	user                 Account
+	hostZone             stakeibctypes.HostZone
+	zoneAccount          Account
+	initialState         RedeemStakeState
+	validMsg             stakeibctypes.MsgRedeemStake
+	expectedNativeAmount sdkmath.Int
+}
+
+func (s *KeeperTestSuite) SetupRedeemStake() RedeemStakeTestCase {
```

Comment:
Note: didn't diff this test but assuming it's a direct copy paste from the msg_server.

Reply:
yeah [this](https://github.com/Stride-Labs/stride/pull/1215/commits/f8148e63a262d486f2b97c9a359e1dab9f626a52) commit was just a copy paste